### PR TITLE
GOINGTOSAI: SWI-4553 - Prepare saiwred.py file for upstreaming

### PR DESCRIPTION
**Pull request by kmalek-priv**

### CODE REVIEW

File: ptf/saiwred.py [hunk 1]
```diff
@@ -0,0 +1,1157 @@
...
+    def _verifyStat(self, i_stats, e_stats, stat, delta):
+        self.assertEqual(e_stats[stat]-i_stats[stat], delta)
+
+    def _verifyStats(self, i_stats, e_stats, *stats):
+        print("Verify stats:")
+        for (name, delta) in stats:
+            print("|", name, "=>", delta)
+            self._verifyStat(i_stats, e_stats, name, delta)
+
+
```
Comment (kmalek-priv) [inline ptf/saiwred.py:252]: This test (traffic test) fails (on model), packet is not ECN-marked as expected (packet verify fails). I do not know why. As I see in jenkins file the test is excluded from the CI scope and now I am not sure if I should remove the test - exclude from the scope to OCP (at least before understanding/fixing failure behavior) or just leave it as is. What do you suggest?

Comment (kperumalbfn) [reply ptf/saiwred.py:252]: Model doesn't have TM which provides the queue_depth. Queue depth is used for WRED meters to determine if the packet can be ECN marked or dropped. Are we validating these tests in hardware switch??

File: ptf/saiwred.py [hunk 2]
```diff
@@ -0,0 +1,431 @@
...
+        print("WRED tests")
+        print("==========\n")
+
+        super(WredTest, self).setUp()
+
+    def runTest(self):
+        self._WredBindProfileTest()
+        self._WredReplaceProfileTest()
+        self._WredBindProfileMultipleQueuesTest()
+        self._WredIPv4NonECNTest()
```
Comment (aljer) [inline ptf/saiwred.py:269]: Please remove underscores from the test names

File: ptf/saiwred.py [hunk 3]
```diff
@@ -0,0 +1,431 @@
...
+        1. Call SaiHelper setUp()
+        2. Configure routing.
+           - route traffic destined for 172.20.20.11 into port 11
+           - route traffic destined for 172.20.20.12 into port 12
+           - route traffic destined for 4000::2 into port 13
+        3. Get queues so that they can be used within tests
+           - self.queues10 of port 10
+           - self.queues11 of port 11
+           - self.queues12 of port 12
+           - self.queues13 of port 13
```
Comment (kperumalbfn) [inline ptf/saiwred.py:41]: We also need to configure DSCP or PCP to TC and TC to queue mapping to configure WRED profiles to non-default queues. Please add them to the setup

File: ptf/saiwred.py [hunk 4]
```diff
@@ -0,0 +1,431 @@
...
+        try:
+            pkt1 = simple_tcp_packet(
+                eth_dst='00:77:66:55:44:00',
+                eth_src='00:22:22:22:22:22',
+                ip_dst='172.20.10.11',
+                ip_src='192.168.0.1',
+                ip_id=105,
+                ip_ttl=64)
+            # ECN unaffected packet
+            exp_pkt1 = simple_tcp_packet(
```
Comment (kperumalbfn) [inline ptf/saiwred.py:398]: Please add v6 traffic tests as well

**Pull request closed without merging**